src/adapter/src/coord/peek.rs

```diff
@@ -1255,12 +1254,14 @@ impl crate::coord::Coordinator {
         .await
     }
 
-    /// Implements a COPY TO command by installing peek watch sets, validating S3 connection,
+    /// Implements a `COPY TO` command by installing peek watch sets,
     /// shipping the dataflow, and spawning a background task to wait for completion.
     /// This is called from the command handler for ExecuteCopyTo.
     ///
-    /// This method inlines the logic from peek_copy_to_preflight and peek_copy_to_dataflow
-    /// to avoid the complexity of the staging mechanism for the frontend peek sequencing path.
+    /// (The S3 preflight check must be completed successfully via the
+    /// `CopyToPreflight` command _before_ calling this method. The preflight is
+    /// handled separately to avoid blocking the coordinator's main task with
+    /// slow S3 network operations.)
```

Comment:
Can we use the `Staged` mechanic to make it harder to hold this wrong? It's how we usually move things off the sequencer thread. Is there a reason it doesn't work here?

Reply:
My issue is mostly about introducing new ways of doing things without removing the old ways ([relevant xkcd](https://xkcd.com/927/)). The individual uses might be simpler, but the overall thing becomes harder to understand because now the code is less consistent. If we had a clear plan to get rid of all the `Staged` stuff, and someone actively working on that, that would be different. But as things stand there is a good chance that the existing `Staged` uses will be around for years to come.

Also, I'm not convinced this way is simpler. At least with `Staged` you don't have to write this comment because the required order is encoded in a state machine. And breaking the state machine would likely get Rust to complain, whereas nothing complains when you ignore the comment.